src/main/java/com/ericsson/ei/frontend/EIRequestsController.java

```diff
@@ -155,36 +169,80 @@
         return getResponse(eiRequest);
     }
 
-    private String getEIBackendSubscriptionAddress() {
-        String httpMethod = "http";
-        if (backEndInformation.isUseSecureHttpBackend()) {
-            httpMethod = "https";
-        }
-
-        if (backEndInformation.getPath() != null && !backEndInformation.getPath().isEmpty()) {
-            return httpMethod + "://" + backEndInformation.getHost() + ":" + backEndInformation.getPort() + "/"
-                    + backEndInformation.getPath();
-        }
-        return httpMethod + "://" + backEndInformation.getHost() + ":" + backEndInformation.getPort();
-    }
-
     private String getEIRequestURL(HttpServletRequest request) {
         String eiBackendAddressSuffix = request.getServletPath();
-        String requestUrl;
+        String requestQuery = request.getQueryString();
+        String requestUrl = null;
+
+        if (requestQuery != null && requestQuery.contains(BACKEND_KEY_NAME)) {
+            // Selecting back end from user input as parameter.
+            List<NameValuePair> params = getParameters(requestQuery);
+            requestUrl = extractUrlFromParameters(params);
+            requestQuery = removeBackendDataFromQueryString(params);
+        } else {
+            BackEndInformation backEndInformation = getEIBackendInformation(request);
+        	requestUrl = backEndInformation.getUrlAsString();
+        }
+
         if(REQUESTS_WITH_QUERY_PARAM.contains(eiBackendAddressSuffix)) {
-            String requestQuery = request.getQueryString();
             String query = (requestQuery != null && !requestQuery.isEmpty()) ? "?" + requestQuery : "";
-            requestUrl = getEIBackendSubscriptionAddress() + eiBackendAddressSuffix + query;
+            requestUrl = requestUrl + eiBackendAddressSuffix + query;
```

Comment:
Shouldn't "eiBackendAddressSuffix" be surrounded by backslashes? or is it required to provide the backslashes in the property files.
I think its better to a one class for whole EI code where you fetch the url arddress. And here you just add the query parameter to the getEiBackendAddress() return value.

Reply:
This change should be made to all places where we need to use the EI backend url.